src/plone/restapi/services/users/get.py

```diff
@@ -46,21 +50,67 @@ def _sort_users(users):
         )
         return users
 
+    def _principal_search_results(
+        self, search_for_principal, get_principal_by_id, principal_type, id_key
+    ):
+
+        hunter = getMultiAdapter((self.context, self.request), name="pas_search")
+
+        principals = []
+        for principal_info in search_for_principal(hunter, self.search_term):
+            principal_id = principal_info[id_key]
+            principals.append(get_principal_by_id(principal_id))
+
+        return principals
+
     def _get_users(self):
         results = {user["userid"] for user in self.acl_users.searchUsers()}
         users = [self.portal_membership.getMemberById(userid) for userid in results]
         return self._sort_users(users)
 
-    def _get_filtered_users(self, query, groups_filter, limit):
-        results = self.acl_users.searchUsers(id=query, max_results=limit)
-        users = [
-            self.portal_membership.getMemberById(user["userid"]) for user in results
-        ]
+    def _user_search_results(self):
+        def search_for_principal(hunter, search_term):
+            return merge_search_results(
+                chain(
+                    *(
+                        hunter.searchUsers(**{field: search_term})
+                        for field in ["name", "fullname", "email"]
+                    )
+                ),
+                "userid",
+            )
+
+        def get_principal_by_id(user_id):
+            mtool = getToolByName(self.context, "portal_membership")
+            return mtool.getMemberById(user_id)
+
+        return self._principal_search_results(
+            search_for_principal, get_principal_by_id, "user", "userid"
+        )
+
+    def _get_filtered_users(self, query, groups_filter, search_term, limit):
+        """Filter or search users by id, fullname, email and/or groups.
+
+        Args:
+            query (str): filter by query
+            groups_filter (list of str): list of groups
+            search_term (str): search by id, fullname, email
+            limit (integer): limit result
+
+        Returns:
+            list: list of users sorted by fullname
+        """
+        if search_term:
+            users = self._user_search_results()
+        else:
+            results = self.acl_users.searchUsers(id=query, max_results=limit)
```

Comment:
@ksuess I tested this [PR](https://github.com/plone/volto/pull/3349) with a ldap backend and found that the query [here](https://github.com/plone/plone.restapi/pull/1444/files#diff-04e041a4f39e375b78c8add4e71438d9d78988fd9c5e30e69cbbd51e9d09553aR28) parses as a list instead of string and It turns an erroneous statement while searching for users with `many_users`  flag checked.

Could you please verify? Thank you.

Reply:
The error I got was 
```
replacement index 1 out of range for positional args tuple
```
It turns out that parse_qs returns a dict https://docs.python.org/3/library/urllib.parse.html#urllib.parse.parse_qs